flink-dist/src/main/flink-bin/bin/bash-java-utils.sh

```diff
@@ -25,11 +34,41 @@ readFromConfigFile() {
     # first extract the value with the given key (1st sed), then trim the result (2nd sed)
     # if a key exists multiple times, take the "last" one (tail)
     local value=`sed -n "s/^[ ]*${key}[ ]*: \([^#]*\).*$/\1/p" "${configFile}" | sed "s/^ *//;s/ *$//" | tail -n 1`
-
+    if [ -z "$value" ]; then
+		# for env.java.home value = `sed -n "/^env:/,/^[^ ]/p" "${configFile}" | sed -n "/^  java:/,/^[^ ]/p" | sed -n "/home:/s/.*: //p" | tail -n 1`
+		IFS='.'
+		read -ra nodes <<< "$1"
+		
+		local sed_cmd_left="sed -n \"/^"
+		local sed_cmd_right=" | "		
+		for index in ${!nodes[@]}; do
+			node=${nodes[index]}
+		 		 
+			if [ $index -eq 0 ]; then
```

Comment:
I suggest adding more comments explaining the intent of the code; it would be easier to parse for the reader

Reply:
get it,  I will follow your above suggest; I have no good idea to add  tests for this now, I just test new logic is adaptive for "flink-conf.yaml" and "config.yaml", I will try add to tests, also looking forward to your good idea